app/controllers/users/piv_cac_recommended_controller.rb

```diff
@@ -0,0 +1,58 @@
+module Users
+  class PivCacRecommendedController < ApplicationController
+    include TwoFactorAuthenticatableMethods
+    include MfaSetupConcern
+    include SecureHeadersConcern
+    include ReauthenticationRequiredConcern
+
+    before_action :authenticate_user!
+    before_action :confirm_user_authenticated_for_2fa_setup
+    before_action :apply_secure_headers_override
+    before_action :user_email_is_gov_or_mil?
+
+    helper_method :in_multi_mfa_selection_flow?
+
+    def show
+      @email_type = email_type
+      @skip_text = skip_text
+      analytics.piv_cac_recommended_page_visited
+    end
+
+    def confirm
+      UpdateUser.new(user: current_user, attributes: { piv_cac_recommended_dismissed: true }).call
+      user_session[:mfa_selections] = ['piv_cac']
```

Comment:
It'd be nice if we could abstract this somehow vs. manipulating the session. Maybe a helper method in `MfaSetupConcern` ?

Reply:
I manipulated the session so that users that are going through account creation flow are properly prompted to add another method after they are done adding piv, but open to other ideas.

Reply:
I don't necessarily think the session manipulation itself is bad, just that we could abstract it in a way that helps protect against potential hazards of working with the raw values, that we could test in isolation, and that hides implementation details in case we wanted it to work differently in the future.

e.g.

```rb
module MfaSetupConcern
  # ...
  def set_mfa_selections(selections)
  # or? def mfa_selections=(selections)
    user_session[:mfa_selections] = selections
  end
end
```

It's also pretty simple so not overly concerned at the moment.